docs/docs/api/producer.md

```diff
@@ -10,15 +10,22 @@ import { Producer } from "@c11/engine.macro"
 
 ## Overview
 
-A `Producer` is a function that reacts to specific state changes and produces new data through specific state updates. It can also produce effects. 
+A `Producer` is simply a function that gets executed when something interesting
+(a trigger) changes in state.
+
+Any function can be turned into a producer by labeling it with `producer`. It
+allows using Engine features to interact with the state using
+[Observe](/docs/api/observe), [Get](/docs/api/get), and
+[Update](/docs/api/update).
 
```

Comment:
Also `Prop` is important. When a producer is added to a view, the producer now receives the same information from the view's parent as the view does, This is useful when you want to sync views->producers to the same data